crates/aeronet_websocket/src/server/config.rs

```diff
@@ -6,9 +6,80 @@ use {
     core::net::{Ipv6Addr, SocketAddr},
     derive_more::{Display, Error},
     rustls::pki_types::{CertificateDer, PrivateKeyDer},
-    tokio_tungstenite::tungstenite::protocol::WebSocketConfig,
+    tokio_tungstenite::tungstenite::{
+        handshake::server::{ErrorResponse, Request, Response},
+        protocol::WebSocketConfig,
+    },
 };
 
+/// Allows inspecting the handshake [`Request`] and conditionally accepting or
+/// denying it based on something like the header.
+///
+/// Useful for implementing something like netcode connect tokens to
+/// authenticate clients before allocating resources for a WS connection.
+#[derive(Clone)]
+pub struct HandshakeHandler(
+    Arc<dyn Fn(&Request, Response) -> Result<Response, ErrorResponse> + Send + Sync>,
+);
+
+impl HandshakeHandler {
+    /// Create a handshake validator to accept/reject incoming ws connections
+    /// during the upgrade handshake.
+    ///
+    /// - Stateful logic is possible with Arc<RwLock<..>> or similar.
+    /// - The Response can be modified before returning.
```

Comment:
doc nits
- types should be wrapped in backquotes
- Response is not a doclink to a type here, so lowercase it
```suggestion
    /// - You can read and write Bevy state by passing shared-mutable-state primitives like `Arc<RwLock<..>>` or MPSC channels into the handler.
    /// - The response can be modified before returning.
```